src/main/java/io/github/syst3ms/skriptparser/expressions/ExprElement.java

```diff
@@ -103,7 +103,7 @@ public boolean isSingle() {
 					case 1:
 						return new Object[] {values[values.length - 1]};
 					case 2:
-						return new Object[] {values[NumberMath.random(0, values.length, true, random).intValue()]};
+						return new Object[] {values[NumberMath.random(0, values.length - 1, true, random).intValue()]};
```

Comment:
After writing my last reply I realized that this also causes an issue (or at least undefined behavior) when the array is empty.
I suggest the following fix:
```suggestion
						return new Object[] {values[NumberMath.random(0, Math.max(values.length - 1, 0), true, random).intValue()]};
```

Reply:
This is not necessary as there is already a check for that scenario.